internal/db/schema/postgres/postgres.go

```diff
@@ -323,33 +320,44 @@ func (p *Postgres) setVersion(ctx context.Context, version int, dirty bool) erro
 	return nil
 }
 
-// CurrentState returns the version, if the database is currently in a dirty state, and any error.
-// A version value of -1 indicates no version is set.
-func (p *Postgres) CurrentState(ctx context.Context) (version int, dirty bool, err error) {
+// CurrentState returns the version, if the database was ever initialized
+// previously, if it is currently in a dirty state, and any error. A version
+// value of -1 indicates no version is set.
+func (p *Postgres) CurrentState(ctx context.Context) (version int, previouslyRan, dirty bool, err error) {
 	const op = "postgres.(Postgres).CurrentState"
-	query := `SELECT Version, dirty FROM ` + pq.QuoteIdentifier(defaultMigrationsTable) + ` LIMIT 1`
+
+	tableQuery := `select table_name from information_schema.tables where table_schema=(select current_schema()) and table_name in ('schema_migrations', '` + defaultMigrationsTable + `')`
+	tableName := defaultMigrationsTable
+	if err = p.conn.QueryRowContext(ctx, tableQuery).Scan(&tableName); err != nil {
+		if err == sql.ErrNoRows {
+			return nilVersion, false, false, nil
+		}
+		return nilVersion, false, false, errors.Wrap(err, op)
+	}
+
+	query := `select version, dirty from ` + pq.QuoteIdentifier(tableName) + ` limit 1`
```

Comment:
This hides if more than one row exists. I think we should guard against this just to be safe.